src/archive/src/main.rs

```diff
@@ -239,6 +259,84 @@ fn write_entry(anchor: AnchorNumber, timestamp: Timestamp, entry: ByteBuf) {
             ))
         }
     });
+    write_entry_internal(anchor_number, timestamp, entry)
+}
+
+/// Fetches, archives and acknowledges a batch of entries.
+/// *Note:* Must be written in a way that nothing breaks on overlapping executions of [fetch_entries].
+async fn fetch_entries() {
+    let ii_canister = with_config(|config| config.ii_canister);
+    let fetch_result: CallResult<(Vec<BufferedEntry>,)> =
+        call(ii_canister, "fetch_entries", ()).await;
+
+    let entries = match fetch_result {
+        Ok((entries,)) => entries,
+        Err((code, message)) => {
+            // failed to fetch entries --> store failure information and exit early
+            store_call_error(CallErrorInfo {
+                time: time(),
+                canister: ii_canister,
+                method: "fetch_entries".to_string(),
```

Comment:
I can see this and the method used in `call` above getting out of sync very quickly

Reply:
Good point.